Relationships/RelationshipResolver.cs

```diff
@@ -397,12 +441,29 @@ public string ReasonStringFor(Module mod)
     /// </summary>
     internal abstract class Relationship
```

Comment:
Is there a chance we can get this called a different name? `Relationship` to me immediately brings to mind conflicts/depends/suggests/recommends, rather than pre-installed/user-requested/due-to-dependency. I don't want to overload our English words *too* much. :)

`ResolveState` doesn't feel quite right. `InstallState` is a little better. `ModuleState` feels to ambiguous. This is this a subclass (which it looks like it is), then `SelectionReason` (or something similar) feels good. Outside this class that would be seen as `RelationshipResolver.SelectionReason`, which is unambiguous. Inside the class `SelectionReason` makes perfect sense as a shorthand.